### PR TITLE
Add logging around `_post_process_results` recursion

### LINKED ISSUE #3189: Add logging & tallying for deep pagination
## Description

We currently don't have much insight into how many iteration our dead link filtering must go through in order to serve requests. We believe that deeper pagination (e.g. iterating over lots and lots of dead links) may be causing our recent increased API response times. We would like to add some logging and tallying into the dead link filtering iterations to know how deep we have to go for certain requests. There's a few steps  we could handle this:

- Logging a query that requires iterating more than `n` times (5? 10?) through the dead page logic, with the query parameters
- Tallying the above, along with a date * time window (e.g. have "number of requests that paginated more than 5 pages in a 5 minute window" as a tally we can observe, this might also require tallying _all_ requests in said 5 minute window that hit the search endpoint, so we can get a ratio of dead link iteration to regular requests)

It would also be ideal if we could set up a sample dashboard or Log Insights query for binning the number of requests which go more than `n` pages deep per 5 minutes from the logs, that way we could observe it over time.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #3189 by @AetherUnbound 

## Description

This PR
- writes log messages when `_post_process_results` recurses more than 5 levels deep (configurable via env var)
- refactors `_post_process_results` to make it responsible for 1 task

## Testing Instructions

The CI includes a test case that captures the logs and verifies that the log line is emitted.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by sarayourfriend — APPROVED** [review]
LGTM! Just a couple of nits, nothing major.

File: api/api/views/media_views.py [hunk 1]
```diff
@@ -83,7 +83,9 @@ def get_db_results(self, results):
         results = list(self.get_queryset().filter(identifier__in=identifiers))
         results.sort(key=lambda x: identifiers.index(str(x.identifier)))
         for result, hit in zip(results, hits):
-            result.fields_matched = getattr(hit, "fields_matched", None)
+            result.fields_matched = None
+            if hl := getattr(hit.meta, "highlight", None):
+                result.fields_matched = hl
```
[the remark this answers is not in the document]
Comment (obulat) [reply api/api/views/media_views.py:88]: Should the default be `[]` instead of `None`? So that the API always returns a list, even if empty?

**Review by obulat — APPROVED** [review]
Clean addition, great tests! Since we have several frontend issues where some property is undefined instead of an expected `list`, I've added a suggestion on using an empty list for highlights when there are none.

File: api/api/views/media_views.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhruvkb) [reply api/api/views/media_views.py:88]: @obulat the serializer converts `None` to `[]`.

File: api/api/controllers/search_controller.py [hunk 2]
```diff
@@ -68,15 +74,23 @@ def _post_process_results(
     :param filter_dead: Whether images should be validated.
```
[the remark this answers is not in the document]
Comment (dhruvkb) [reply api/api/controllers/search_controller.py:74]: Added in 4e2f6d4.

**Pull request merged**